### DISCUSSION TITLE
TrustProxies issue

### DISCUSSION THREAD

**Discussion by pratikshap31** [enlightn/enlightn#122] [General]
How to fix Your application is vulnerable to a host injection attack in aws server 
i have try with two option:
    - using TrustHosts 
    - using TrustedProxy
Here is my code using TrustProxies Middleware :
    app/Http/Middleware/TrustProxies.php
    <?php

    namespace App\Http\Middleware;

    use Illuminate\Http\Middleware\TrustProxies as Middleware;
    use Illuminate\Http\Request;

    class TrustProxies extends Middleware
    {
        /**
         * The trusted proxies for this application.
         *
         * @var array|string|null
         */

        protected $proxies = ['MyIP', '*'];

        /**
         * The headers that should be used to detect proxies.
         *
         * @var int
         */
        protected $headers = Request::HEADER_X_FORWARDED_FOR | Request::HEADER_X_FORWARDED_AWS_ELB;
        // protected $headers = ;
    }

    Here is my code using TrustHosts Middleware :
    app/Http/Middleware/TrustHosts.php
    <?php
        namespace App\Http\Middleware;
        use Illuminate\Http\Middleware\TrustHosts as Middleware;

        class TrustHosts extends Middleware
        {
            /**
             * Get the host patterns that should be trusted.
             *
             * @return array
             */
            public function hosts()
            {
                return [
                    'mydomain',
                    $this->allSubdomainsOfApplicationUrl(),
                ];
            }
        }

**Comment by paras-malhotra** [comment]
To fix the issue, please refer the options described in the docs here: https://www.laravel-enlightn.com/docs/security/host-injection-analyzer.html#how-to-fix-host-header-injection

**Reply by pratikshap31** [reply]
@paras-malhotra 
I have already tried that steps.
Can you please provide steps?

**Reply by paras-malhotra** [reply]
Which option have you tried? There are 4 options

**Reply by pratikshap31** [reply]
@paras-malhotra 

Option 1 and 2
option 3 and 4 are for  nginx server. we are using apache2.

**Reply by paras-malhotra** [reply]
Do you have both `TrustHosts` and `TrustProxies` in your middleware?

**Reply by pratikshap31** [reply]
@paras-malhotra 

Yes. We have both middleware.

**Comment by paras-malhotra** [comment]
@pratikshap31 I just discovered that the `TrustProxies` middleware was moved to `Illuminate\Http\Middleware` in Laravel 9. Earlier it was part of the `fideloper/proxy` package.

This is now fixed in Enlightn Pro v2.2.0. You can upgrade to that version and the error should be fixed. Also, you only need to do either option 1 or option 2. Both are not required, but it's perfectly fine if you wish to do both.

**Reply by pratikshap31** [reply]
@paras-malhotra 
Can we have any other option?
Because my project is set up with the current version, so, when I have tried to update it returns errors.

**Comment by pratikshap31** [comment]
It still not working for me.

**Reply by paras-malhotra** [reply]
Are you using Enlightn Pro v2.2.0?

**Reply by pratikshap31** [reply]
No. i am using Enlightn Pro v1.*

**Reply by paras-malhotra** [reply]
It has been fixed in Enlightn Pro v2.2.0. You will have to upgrade your composer to that version.

**Reply by pratikshap31** [reply]
I am using laravel 8 and have some packages that support <= laravel 8. and enlight 2.2 does not support laravel 8. do we have any other solution?
